merlin/models/tf/inputs/embedding.py

```diff
@@ -1030,24 +1030,19 @@ def get_config(self):
             feature_configs[key] = feature_config_dict
 
         config["feature_config"] = feature_configs
+        config["l2_reg"] = self.l2_reg
 
         return config
 
     @classmethod
     def from_config(cls, config):
         # Deserialize feature_config
-        feature_configs, table_configs = {}, {}
+        feature_configs = {}
         for key, val in config["feature_config"].items():
-            feature_params = deepcopy(val)
-            table_params = feature_params["table"]
-            if "name" in table_configs:
```

Comment:
This is the line that was causing issues. Once a features called `"name"` is processed. It will skip all remaining features and assign them to share the name feature. Resulting in the embeddings for the rest of the features having the same embedding as the `name` feature.